controller/workitem.go

```diff
@@ -111,18 +136,44 @@ func (c *WorkitemController) Update(ctx *app.UpdateWorkitemContext) error {
 	if !authorized {
 		return jsonapi.JSONErrorResponse(ctx, errors.NewForbiddenError("user is not authorized to access the space"))
 	}
+
+	if ctx.Payload.Data.Relationships != nil && ctx.Payload.Data.Relationships.BaseType != nil &&
+		ctx.Payload.Data.Relationships.BaseType.Data.ID != wi.Type {
+		// Store new values of type and version
+		newTypeID := ctx.Payload.Data.Relationships.BaseType
+		newVersion := ctx.Payload.Data.Attributes[workitem.SystemVersion]
+
+		// Remove version and base type from payload
+		delete(ctx.Payload.Data.Attributes, workitem.SystemVersion)
+		ctx.Payload.Data.Relationships.BaseType = nil
+
+		// Ensure we do not have any other change in payload except type change
+		if (app.WorkItemRelationships{}) != *ctx.Payload.Data.Relationships || len(ctx.Payload.Data.Attributes) > 0 {
+			return jsonapi.JSONErrorResponse(ctx, errors.NewBadParameterErrorFromString("cannot update type along with other fields"))
```

Comment:
@jarifibrahim This case should not throw a Bad Parameter Error. The request is syntactically well-formed, but semantically not correct. So[ **422 Unprocessable Entity** ](https://tools.ietf.org/html/rfc4918#section-11.2)HTTP status code should be used here.

Reply:
@DhritiShikhar @kwk Do you want this change to be part of this PR? I will have to update the `fabric8-wit/errors` package and related tests. This might add around 100 more changes to this PR.

Reply:
I've added a `TODO` here https://github.com/fabric8-services/fabric8-wit/pull/2202/commits/f19282efe6488ed76cfe4f2729de0352ad69e875 .